themes/default/content/blog/cloud-systems-part-four/index.md

```diff
@@ -0,0 +1,171 @@
+
+<!--more-->
+
+## Elastic Kubernetes Service
+
+In the previous tutorial, we extended our personal website to use the Flask web framework, add server-side routing, and package everything up into a Docker container. We then used Pulumi to deploy it to AWS Elastic Container Service. What if we want to go bigger, though? Maybe we're expecting a ton of traffic to the site. Let's make it a little more resilient by upgrading from Elastic Container Service to Kubernetes.
+
+If you aren't familiar with Kubernetes, it's a container orchestration tool. Tools like this are responsible for the management, deployment, and scaling of containerized applications. It's possible to do all of that by hand, of course, but it's slow, difficult, and unreliable. If you're just now joining me, you can get the completed code by forking and cloning [this repository](https://github.com/katcosgrove/cloud-systems-101).
+
+## Prerequisites:
+
+- An AWS Account
+
+- [Pulumi account](https://app.pulumi.com)
+
+- [Pulumi installed and configured for AWS](https://www.pulumi.com/docs/get-started/aws/begin/)
+
+- [Docker](https://www.docker.com/products/docker-desktop)
+
+- Python3
+
+## Crosswalk, EKS, and Pulumi
+
+We'll be re-using some things we did in the previous tutorials, but we do need some new dependencies. Pulumi Crosswalk is a collection of libraries for interacting with AWS while adhering to common best practices, without having to do most of the work yourself. It's great for getting off the ground with some of the most common first-step tasks when deploying infrastructure on AWS, so we'll be using it. We also need providers for AWS EKS and Kubernetes. The top of your `__main__.py` should include the following imports:
+
+```python
+import base64
+import pulumi
+import pulumi_awsx as awsx
+import pulumi_eks as eks
+import pulumi_kubernetes as k8s
+```
+
+Make sure you install all of the new things:
+
+```bash
+pip3 install pulumi_awsx pulumi_eks pulumi_kubernetes
+```
+
+Now we're ready to build some things. First, we want to create an EKS cluster and an ECR repository. We created an used an ECR repository in part three of this series, but it took quite a bit of code. Since we're using Crosswalk to handle a lot of those default settings for us this time, we don't have to do as much work.
```

Comment:
Link to part 3?
```suggestion
Now we're ready to build some things. First, we want to create an EKS cluster and an ECR repository. We created and used an ECR repository in part three of this series, but it took quite a bit of code. Since we're using Crosswalk to handle a lot of those default settings for us this time, we don't have to do as much work:
```